lib/good_job/active_job_extensions/logging.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+module GoodJob
+  module ActiveJobExtensions
+    module Logging
+      extend ActiveSupport::Concern
+
+      included do
+        around_perform do |job, block|
+          original_logger = job.logger
+          job.logger = ActiveSupport::TaggedLogging.new(ActiveSupport::Logger.new(LogDevice.new(job)).extend(ActiveSupport::Logger.broadcast(original_logger)))
+          block.call
+        ensure
+          job.logger = original_logger
+        end
+      end
+
+      class LogDevice
+        cattr_accessor :logs, default: Concurrent::Array.new
```

Comment:
@bensheldon is the `c` in `cattr_accessor` to indicate that this is class attribute

Reply:
Yes, it's a class attribute. By living on the class it becomes a singleton that can be access globally.